frontend/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -330,6 +333,68 @@ export class WorkflowEditorComponent implements OnInit, AfterViewInit, OnDestroy
       });
   }
 
+  private handleRegion(): void {
+    this.editor.classList.add("hide-region");
+    const Region = joint.dia.Element.define(
+      "region",
+      {
+        attrs: {
+          body: {
+            fill: "rgba(255,213,79,0.2)",
+            stroke: "#FFD54F",
+            "stroke-width": 3,
+            "stroke-dasharray": "6,4",
+            pointerEvents: "none",
+            class: "region",
+          },
+        },
+      },
+      {
+        markup: [{ tagName: "path", selector: "body" }],
+      }
+    );
+
+    let regionMap: { regionShape: joint.dia.Element; operators: joint.dia.Cell[] }[] = [];
+    // update region shapes on execution
```

Comment:
The name "shape" is very confusing. You are referring to a region's Joint element as a shape, but when an operator is moved, you actually also change the "shape" (i.e., contour) of the region element. Why not just  name them something like `RegionElement` or `RegionJointElement` to avoid ambiguity?

Reply:
Please also update other places of `shape`.